contracts/token/types/TokenOrEther18.sol

```diff
@@ -0,0 +1,174 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import "../../number/types/UFixed18.sol";
+
+//TODO: add .eq()
+
+/// @dev TokenOrEther18
+type TokenOrEther18 is address;
+using TokenOrEther18Lib for TokenOrEther18 global;
+type TokenOrEther18Storage is bytes32;
+using TokenOrEther18StorageLib for TokenOrEther18Storage global;
+
+/**
+ * @title TokenOrEther18Lib
+ * @notice Library to manage Ether and ERC20s that is compliant with the fixed-decimal types.
+ * @dev Normalizes token operations with Ether operations (using a magic Ether address)
+ */
+library TokenOrEther18Lib {
+    using Address for address;
+    using SafeERC20 for IERC20;
+
+    error TokenOrEther18PullEtherError();
+    error TokenOrEther18ApproveEtherError();
+
+    TokenOrEther18 public constant ZERO = TokenOrEther18.wrap(address(0));
```

Comment:
Why do we need a ZERO value?

Reply:
I added `ZERO` to match the defaults we have in the `Fixed` types.

It's probably not going to be super widely used, but there's some cases where it'd be nice to have like populating defaults in structs or validation checks.